driver-sync/src/test/functional/com/mongodb/client/Socks5ProseTest.java

```diff
@@ -0,0 +1,201 @@
+    @DisabledIf("isAuthEnabled")
+    void shouldConnectWithoutAuthInProxySettings(final ConnectionString connectionString) {
+        mongoClient = MongoClients.create(buildMongoClientSettings(connectionString));
+        runHelloCommand(mongoClient);
+    }
+
+    @ParameterizedTest(name = "Should not connect without valid authentication in connection string. ConnectionString: {0}")
+    @MethodSource({"noAuthConnectionStrings", "invalidAuthConnectionStrings"})
+    @EnabledIf("isAuthEnabled")
+    void shouldNotConnectWithoutAuth(final ConnectionString connectionString) {
+        ClusterListener clusterListener = Mockito.mock(ClusterListener.class);
+        ArgumentCaptor<ClusterDescriptionChangedEvent> captor = ArgumentCaptor.forClass(ClusterDescriptionChangedEvent.class);
+
+        mongoClient = createMongoClient(MongoClientSettings.builder()
+                .applyConnectionString(connectionString), clusterListener);
+
+        Assertions.assertThrows(MongoTimeoutException.class, () -> runHelloCommand(mongoClient));
+        assertSocksAuthenticationIssue(clusterListener, captor);
+    }
+
+    @ParameterizedTest(name = "Should not connect without valid authentication in proxy settings. ConnectionString: {0}")
+    @MethodSource({"noAuthConnectionStrings", "invalidAuthConnectionStrings"})
+    @EnabledIf("isAuthEnabled")
+    void shouldNotConnectWithoutAuthInProxySettings(final ConnectionString connectionString) {
+        ClusterListener clusterListener = Mockito.mock(ClusterListener.class);
+        ArgumentCaptor<ClusterDescriptionChangedEvent> captor = ArgumentCaptor.forClass(ClusterDescriptionChangedEvent.class);
+
+        mongoClient = createMongoClient(MongoClientSettings.builder(buildMongoClientSettings(connectionString)), clusterListener);
+
+        Assertions.assertThrows(MongoTimeoutException.class, () -> runHelloCommand(mongoClient));
+        assertSocksAuthenticationIssue(clusterListener, captor);
+    }
+
+    @ParameterizedTest(name = "Should connect with valid authentication in connection string. ConnectionString: {0}")
+    @MethodSource("validAuthConnectionStrings")
+    @EnabledIf("isAuthEnabled")
+    void shouldConnectWithValidAuth(final ConnectionString connectionString) {
+        mongoClient = MongoClients.create(connectionString);
+        runHelloCommand(mongoClient);
+    }
+
+    @ParameterizedTest(name = "Should connect with valid authentication in proxy settings. ConnectionString: {0}")
+    @MethodSource("validAuthConnectionStrings")
+    @EnabledIf("isAuthEnabled")
+    void shouldConnectWithValidAuthInProxySettings(final ConnectionString connectionString) {
+        mongoClient = MongoClients.create(buildMongoClientSettings(connectionString));
+        runHelloCommand(mongoClient);
+    }
+
+    private static void assertSocksAuthenticationIssue(final ClusterListener clusterListener,
+                                                       final ArgumentCaptor<ClusterDescriptionChangedEvent> captor) {
```

Comment:
`captor` is accepted here, but really it can be created by the `assertSocksAuthenticationIssue` method instead.